### PR TITLE
Use enums for representing the state of a figure

### PR DESCRIPTION
**Pull request by guemax**
Enums provide the benefit of not being prone to accidental changes by a programmer who sets the boolean flags previously used by `Figure` in a wrong manner, similar to this:

```java
figure[i].inHouse = true;
// And somewhere down the same program flow
figure[i].inBase = true;
```

This can be a real problem, because a figure can never be in its house and in its base at the same time!

Since enums either take one value or another, this problem can easily be prevented. It also serves to produce much cleaner code, which I have amplified by moving this logic to the class `Figure`, which should be responsible for managing its own state.

### CODE REVIEW

**Review by MeiNic — APPROVED** [review]
Great job.👍 
The changes make a lot of sense, so the described scenario can certainly not occur, where that was currently prevented by the code. I did not know the data type "enum" until this code review😂. I would still wait for another review

**Review by TastingComb — APPROVED** [review]

**Pull request merged**
